intelmq/bots/outputs/rpz_file/output.py

```diff
@@ -0,0 +1,154 @@
+# -*- coding: utf-8 -*-
+import datetime
+from datetime import datetime
+import os
+from collections import defaultdict
+from pathlib import Path
+
+from intelmq.lib.bot import OutputBot
+
+RPZ_INDICATOR_MAP = {
+    "source.fqdn": "Intel::DOMAIN"
+}
+
+now = datetime.now()
+
+
+class RpzFileOutputBot(OutputBot):
+    _file = None
+    format_filename: bool = False
+    hierarchical_output: bool = False
+    keep_raw_field: bool = False
+    message_jsondict_as_string: bool = False
+    message_with_type: bool = False
+    single_key: bool = False
+    is_multithreadable = False
+
+    encoding_errors_mode = 'strict'
+    file: str = "/opt/intelmq/var/lib/bots/file-output/rpz"
+
+    cname: str = ""
+    organization_name: str = ''
+    rpz_domain: str = ''
+    hostmaster_rpz_domain: str = ''
+    rpz_email: str = ''
+    ttl: str = '3600'
+    generate_time: str = now.strftime("%Y-%m-%d %H:%M:%S")
```

Comment:
Is that a configuration parameter? If yes, why does its default value depend on the time the bot is created / configured?

Reply:
Yes, it is configuration parameter. generate_time it is like timestamp, if regenerate file this value need change.

Reply:
Sorry, I still don't understand.
You say that this is a configuration parameter. It's default value is "now". Let's assume, that a user configures an instance of the RPZ output. The default value is `2021-07-16 18:17:24`, which looks good to the user and sets that. As a result, the generated file always has the exact same timestamp. It could be interpreted as "the time when the user has configured this bot" or "initial creation date".
In the header, this value is used for "Last updated" which has - for me - a different meaning. "Last update" could be the time when the last entry was added to the file.

Reply:
move to function, it is not configuration  parameter now.